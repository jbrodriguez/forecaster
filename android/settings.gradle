rootProject.name = 'itcrowd'

include ':app'
