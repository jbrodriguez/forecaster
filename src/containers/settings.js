// @flow

import React, { PureComponent } from 'react'
import { View, Text } from 'react-native'

import type { Dispatch } from 'redux'
import { connect } from 'react-redux'
import SettingsList from 'react-native-settings-list'
import { NavigationActions } from 'react-navigation'

import type { TAppState, TGui, TTempUnit } from '../typings'
import { getGui, getVersion } from '../modules/env'
import { tempUnits, getTempUnit } from '../modules/settings'

type Props = {
  gui: TGui,
  version: string,
  tempUnit: TTempUnit,
  dispatch: Dispatch,
}

class Settings extends PureComponent<Props> {
  static navigationOptions = {
    title: 'Settings',
  }

  onTemp = () => {
    this.props.dispatch(NavigationActions.navigate({ routeName: 'TempUnit' }))
  }

  render() {
    const { gui: { s, c }, tempUnit, version } = this.props

    return (
      <View style={[s.flx_i, s.pa2, c.g_background]}>
        <View style={[s.flx_row, s.aic, s.mb3]}>
          <Text style={[s.f5, c.c_primary]}>Forecaster version:</Text>
          <Text style={[s.ml2, s.f5, c.c_accent]}>{version}</Text>
        </View>

        <SettingsList backgroundColor={c.background} borderColor={c.divider} defaultTitleStyle={[s.f5, c.c_accent]}>
          <SettingsList.Item
            title="Temperature Unit"
            titleInfo={tempUnits[tempUnit].label}
            titleInfoStyle={[s.f5, c.c_primary]}
            onPress={this.onTemp}
          />
        </SettingsList>
      </View>
    )
  }
}

const mapStateToProps = (state: TAppState) => ({
  gui: getGui(state),
  version: getVersion(state),
  tempUnit: getTempUnit(state),
})
const mapDispatchToProps = dispatch => ({ dispatch })

export default connect(mapStateToProps, mapDispatchToProps)(Settings)
