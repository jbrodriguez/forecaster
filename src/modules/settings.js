// @flow

import ActionKey from '../typings'
import type { TSettingsState, AOther, TAppState, TTempUnit, ASetTempUnit } from '../typings'

// ACTION CREATORS
export const setTempUnit = (payload: TTempUnit): ASetTempUnit => ({ type: ActionKey.SET_TEMP_UNIT, payload })

// REDUCER
export const tempUnits = {
  imperial: { label: 'Fahrenheit', value: 'imperial' },
  metric: { label: 'Celsius', value: 'metric' },
}

const initialState: TSettingsState = {
  tempUnit: 'metric',
}

type TAction = AOther | ASetTempUnit

const reducer = (state: TSettingsState = initialState, action: TAction): TSettingsState => {
  switch (action.type) {
    case ActionKey.SET_TEMP_UNIT:
      return {
        ...state,
        tempUnit: action.payload,
      }

    default:
      return state
  }
}

export default reducer

// SELECTORS
export const getTempUnit = (state: TAppState): TTempUnit => state.settings.tempUnit

// SAGAS
