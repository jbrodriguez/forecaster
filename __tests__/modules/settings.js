import 'react-native'
// import React from 'react'

// Note: test renderer must be required after react-native.
// import renderer from 'react-test-renderer'

import settings, { initialState, setTempUnit } from '../../src/modules/settings'

describe('settings/reducer', () => {
  test('return the same state on unhandled action', () => {
    expect(settings(initialState, { type: '_NULL_' })).toMatchSnapshot()
  })

  test('handles setTempUnit/imperial', () => {
    expect(settings(initialState, setTempUnit('imperial'))).toMatchSnapshot()
  })

  test('handles setTempUnit/metric', () => {
    expect(settings({ ...initialState, tempUnit: 'imperial' }, setTempUnit('metric'))).toMatchSnapshot()
  })
})

describe('settings/action creators', () => {
  test('creates setTempUnit', () => {
    expect(setTempUnit('imperial')).toMatchSnapshot()
  })
})
