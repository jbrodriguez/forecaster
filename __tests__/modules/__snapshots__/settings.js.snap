// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings/action creators creates setTempUnit 1`] = `
Object {
  "payload": "imperial",
  "type": "forecaster/settings/SET_TEMP_UNIT",
}
`;

exports[`settings/reducer handles setTempUnit/imperial 1`] = `
Object {
  "tempUnit": "imperial",
}
`;

exports[`settings/reducer handles setTempUnit/metric 1`] = `
Object {
  "tempUnit": "metric",
}
`;

exports[`settings/reducer return the same state on unhandled action 1`] = `
Object {
  "tempUnit": "metric",
}
`;
